src/main/java/com/fcgl/Listing/Vendors/Amazon/Feed/GenerateFeedSubmissionList/GenerateFeedSubmissionList.java

```diff
@@ -0,0 +1,197 @@
+ */
+public class GenerateFeedSubmissionList  extends AbstractAMZService {
+  private final String requestId;
+  private IMessageQueueSender messageQueueSender;
+  private final int[] retryWaitTime = {1000, 4000, 16000, 30000};//This was recommended by Amazon //TODO: Should put this somewhere else
+  private static final Logger logger = LogManager.getLogger("GenerateFeedSubmissionList");
+  private String PROCESSING_COMPLETE = "ProcessingComplete";
+  private String FEED_CANCELED = "FeedCanceled";
+  private String SAFETY_NET = "SafetyNet";
+  private String FEED_PROCESSING = "FeedProcessing";
+
+
+  public GenerateFeedSubmissionList(String requestId, IMessageQueueSender messageQueueSender) {
+    this.requestId = requestId;
+    this.messageQueueSender = messageQueueSender;
+  }
+
+  /**
+   * Calls the invokeGetFeedSubmissionList and returns a Response based on the List returned by that
+   * method. Also handles the Exception thrown by invokeGetFeedSubmissionList
+   * @return Response
+   */
+  public Response generateFeedSubmissionList(List<String> submissions) {
+    MarketplaceWebService service = instantiateService();
+    GetFeedSubmissionListRequest request = new GetFeedSubmissionListRequest();
+    request.setMerchant(getAmazonConfig().getMWSAuthToken());
+    Response response = null;
+    boolean success = false;
+    int retry = 3;//TODO: I've been doing retry logic in a lot of places and the retry has always been 3... Maybe I should put that in a config somewhere?
+    while (!success && retry != 0) {
+      try {
+        List<String> feedStatus = invokeGetFeedSubmissionList(service, request);
+        String message = String.format("RequestID: %s, Feed Submissions Processed: %d,"
+            + " Status Code: %d", requestId, feedStatus.size(), 200);
+        handleFeedProcessingStatus(feedStatus, submissions);
+        response = new Response(200, requestId, message);
+        success = true;
+      } catch (MarketplaceWebServiceException e) {
+        e.getResponseHeaderMetadata();
+        response = handleErrorResponse(e, "GenerateFeedSubmissionList", requestId);
+        retry--;
+      }
+    }
+
+    return response;
+  }
+
+  /**
+   * TODO: THis method could probably be private. It wouldn't be hard to mock the incomming data...
+   * //TODO: I would like all of these amazon API calls to be the same, but they are different for right now
+   * //TODO: I will come up with a better solution once finishing other tasks.
+   * Calls the getFeedSubmissionList Amazon API. It returns the status for feed submissions.
+   * Calls the handleFeedProcessingStatus() method which adds the feedSubmissionIds to a specified
+   * Queue for more processing.
+   * @param service instance of MarketplaceWebService service
+   * @param request a GetFeedSubmissionListRequest
+   * @return Response Object
+   */
+  private List<String> invokeGetFeedSubmissionList(MarketplaceWebService service, GetFeedSubmissionListRequest request)
```

Comment:
These methods are very ugly. It's very long and there is very little logic in it. It would be nice to abstract this out somehow. Perhaps not in this PR but should be something we think about because I don't want to follow this pattern for every API call we make to Amazon